test/workbox-core/node/utils/test-logger.mjs

```diff
@@ -0,0 +1,303 @@
+import {expect} from 'chai';
+import sinon from 'sinon';
+
+import logger from '../../../../packages/workbox-core/internal/utils/logger.mjs';
+import WorkboxError from '../../../../packages/workbox-core/internal/models/WorkboxError.mjs';
+import generateVariantTests from '../../../../infra/utils/generate-variant-tests';
+
+describe(`logger`, function() {
+  let sandbox;
+
+  before(function() {
+    sandbox = sinon.sandbox.create();
+  });
+
+  beforeEach(function() {
+    // Reset between runs
+    logger.logLevel = logger.LOG_LEVELS.verbose;
+  });
+
+  afterEach(function() {
+    sandbox.restore();
+  });
+
+  describe(`LOG_LEVELS`, function() {
+    it(`should expose the valid LOG_LEVELS`, function() {
+      expect(logger.LOG_LEVELS).to.exist;
+    });
+
+    it(`should expose the expected LOG_LEVELS`, function() {
+      expect(logger.LOG_LEVELS.verbose).to.exist;
+      expect(logger.LOG_LEVELS.debug).to.exist;
+      expect(logger.LOG_LEVELS.warning).to.exist;
+      expect(logger.LOG_LEVELS.error).to.exist;
+    });
+  });
+
+  describe(`.logLevel (setter)`, function() {
+    it(`should allow valid log levels`, function() {
+      expect(() => {
+        logger.logLevel = logger.LOG_LEVELS.verbose;
+        logger.logLevel = logger.LOG_LEVELS.debug;
+        logger.logLevel = logger.LOG_LEVELS.warning;
+        logger.logLevel = logger.LOG_LEVELS.error;
+      }).to.not.throw();
+    });
+
+    it(`should not allow log level less than verbose`, function() {
+      expect(() => {
+        logger.logLevel = logger.LOG_LEVELS.verbose - 1;
+      }).to.throw(WorkboxError).that.has.property('name').that.equals('invalid-value');
+    });
+
+    it(`should not allow log level greater than error`, function() {
+      expect(() => {
+        logger.logLevel = logger.LOG_LEVELS.error + 1;
+      }).to.throw(WorkboxError).that.has.property('name').that.equals('invalid-value');
+    });
+
+    generateVariantTests(`should not allow non-number log levels`, [
+      undefined,
+      null,
+      '',
+      [],
+      {},
+    ], (variant) => {
+      expect(() => {
+        logger.logLevel = variant;
+      }).to.throw(WorkboxError).that.has.property('name').that.equals('invalid-type');
+    });
+  });
+
+  /**
```

Comment:
Minor nit, but this comment block should probably be either `/* ... */` or multiple `// ...`. Using `/**` (with two stars) signifies a JSDoc block, so it's best to not use it unless writing a JSDoc block (this is also required by google JS style, though not enforced by the eslint package).